version: "3.2"

services:
  docker-auth:
    labels:
      traefik.docker.network: traefik_public
      traefik.enable: 'true'
      traefik.frontend.rule: Host:[token.{{domain}}]
      traefik.port: '5001'
    image: cesanta/docker_auth:stable
    build:
      args:
      - "/config/auth_config.yml"
    volumes:
      - {{docker_conf_path}}/Docker-auth/config:/config
      - {{docker_conf_path}}/Docker-auth/logs:/logs
      - {{docker_conf_path}}/Docker-auth/certs:/certs
    deploy:
      replicas: 1
      update_config:
        parallelism: 1
        failure_action: pause
      restart_policy:
        condition: any
        max_attempts: 0
      placement:
        constraints: [node.role == manager]

networks:
  traefik_public:
    external: true
